backend/file-system/src/main/java/com/htec/filesystem/service/FileService.java

```diff
@@ -129,4 +132,19 @@ public void saveFileIntoDB(String filePath, String fileName, long shelfId, long
         fileEntity.setCreatedAt(LocalDateTime.now());
         fileRepository.save(fileEntity);
     }
+
+    @Transactional
+    public void updateDeletedFile(AuthUser user, List<Long> fileIds, boolean delete) {
+        List<FileEntity> fileEntities = fileRepository.findAllByUserIdAndIdIn(user.getId(), fileIds);
+
+        if (fileEntities.size() != fileIds.size()) {
+            throw ExceptionSupplier.fileNotFound.get();
```

Comment:
again, if multiple files are missing, is the exception the most suitable one?

Reply:
Exception message changed to plural